### PR TITLE
Quick attempt at fixing EOF error issue encountered on linux/ARM

### PR DESCRIPTION
**Pull request by elProxy**
When trying to run gcfuse on linux/ARM, ran into the issue described
here:
https://github.com/GoogleCloudPlatform/gcsfuse/issues/197

Change the API slightly (expose even less), and switch to using a local
unix socket (assuming windows is not a primary target for gcsfuse) for
communicating back status to the parent process.

### CODE REVIEW

**Comment by jacobsa** [comment]
Thanks for this. Can you describe here and in your commit message what the problem was?

**Comment by elProxy** [comment]
Well, unfortunately I didn't really dig too deep into why things didn't work, I noticed that things ran fine with --foreground and proceeded to just make this last bit work for me by simply using a unix socket as it feels a bit more flexible for that sort of non performance-critical IPC anyway. Though it's interesting that the other report also happened to be on ARM.

File: daemonize.go [hunk 1]
```diff
@@ -14,30 +14,33 @@
 
 // Helper code for starting a daemon process.
 //
-// This package assumes that the user invokes a tool, which invokes a daemon
-// process. Though the tool starts the daemon process with stdin, stdout, and
-// stderr closed (using Run), the daemon should be able to communicate status
-// to the user while it starts up (using StatusWriter), causing the tool to
+// This package provides the convenience for writing a tool that spins itself
```
Comment (jacobsa) [inline daemonize.go:17]: Why change the wording here?

File: daemonize.go [hunk 2]
```diff
@@ -200,36 +195,29 @@ func Run(
...
+// Start the daemon process and do
 // not wait for it to return.
-func startProcess(
-	path string,
-	args []string,
-	env []string,
-	pipeW *os.File) (err error) {
+func startProcess(args []string, status io.Writer) error {
+
+	// TODO: drop osext dependency once minmum required go verison is 1.8
```
Comment (jacobsa) [inline daemonize.go:202]: Please just do this now. Those binaries that need to build with older versions of Go can vendor an old version.

File: daemonize.go [hunk 3]
```diff
@@ -129,47 +124,42 @@ func (w *logMsgWriter) Write(p []byte) (n int, err error) {
 	return
 }
 
-// For use by the daemon: the writer that should be used for logging status
```
Comment (jacobsa) [inline daemonize.go:132]: Why did you remove this feature?

**Review by jacobsa — CHANGES_REQUESTED** [review]
Sending comments on specifics, but I'm not sure I'm comfortable with making this change without knowing why (and therefore whether) it fixed the root cause of the bug.

**Comment by kazazes** [comment]
This is still a blocking issue on ARM builds.

```
$ go get -u github.com/googlecloudplatform/gcsfuse
# github.com/googlecloudplatform/gcsfuse/vendor/github.com/jacobsa/fuse/internal/buffer
./runtime.go:24: missing function body for "memclr"
./runtime.go:24: missing function body for "memmove"
```

**Pull request closed without merging**